dlt/destinations/impl/lancedb/utils.py

```diff
@@ -0,0 +1,53 @@
+import os
+import uuid
+from typing import Sequence, Union, Dict
+
+from dlt.common.schema import TTableSchema
+from dlt.common.schema.utils import get_columns_names_with_prop
+from dlt.common.typing import DictStrAny
+from dlt.destinations.impl.lancedb.configuration import TEmbeddingProvider
+
+
+PROVIDER_ENVIRONMENT_VARIABLES_MAP: Dict[TEmbeddingProvider, str] = {
+    "cohere": "COHERE_API_KEY",
+    "gemini-text": "GOOGLE_API_KEY",
+    "openai": "OPENAI_API_KEY",
+    "huggingface": "HUGGINGFACE_API_KEY",
+}
+
+
+def generate_uuid(data: DictStrAny, unique_identifiers: Sequence[str], table_name: str) -> str:
```

Comment:
this code seems to be a copy from the other vector DBs, generally speaking if this happens, we should extract it into a utils class that is used by all destinations, for now let's keep it, I think some refactoring is due at some point in any case.